thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/algorithm/DimensionWrapper.java

```diff
@@ -238,6 +244,16 @@ public DetectionPipelineResult run() throws Exception {
           this.config.getId(), nestedMetrics.size(), MAX_DIMENSION_COMBINATIONS));
     }
 
+    // don't use in-memory cache for dimension exploration, it will cause thrashing
+    if (CacheConfig.useCentralizedCache()) {
+      if (this.cachingPeriodLookback >= 0) {
+        this.provider.fetchTimeseries(nestedMetrics.stream()
```

Comment:
Are we doing cache warm-up here? If yes, can this call be slow?

Reply:
Yes, it's used to prefetch data for the detection and warm up the cache. It can be slow depending on the number of dimension combinations we need to make.